source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,189 @@
+configure the method to return the document after the update occurs. For
+more information on the ``findOneAndUpdate()`` method, see our Compound
+Operations guide <TODO>. 
+
+Specifying an Update
+--------------------
+
+To specify an update, use the ``Updates`` builder. The ``Updates``
+builder provides static utility methods to construct update
+specifications. For more information on using the ``Updates`` builder with
+arrays, see our :doc:`guide on the Updates builder </fundamentals/builders/updates>`.
+
+The following example performs these actions: 
+
+- Query for the sample document
+- Add "17" to the ``qty`` array using the ``push()`` method of the ``Updates`` builder
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin pushElementsExample
+   :end-before: end pushElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "color": "green", "qty": [8, 12, 18, 17] }
+
+Specifying Array Elements
+-------------------------
+
+You can specify which array elements to update using a positional
+operator. Positional operators can specify the first, all or certain
+array elements to update. 
+
+To specify elements in an array with positional operators, use the **dot
+notation**. Dot notation is a property access syntax for navigating JSON
+objects. 
+
+For additional information, see the server manual entry on
+:manual:`dot notation </core/document/#std-label-document-dot-notation>`.
+
+The First Array Element
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To update the first element that matches your query filter, use the
+positional ``$`` operator. The array field must appear as part of your
+query document to use the positional ``$`` operator. 
+
+Example
+```````
+
+The following example performs these actions:
+
+- Query for a document with a ``qty`` field containing the value "18"
+- Decrement the first array value that matches the query document by "3"  using the ``inc()`` method of the ``Updates`` builder and a positional ``$`` operator 
+- Update and retrieve the document
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin updateValueExample
+   :end-before: end updateValueExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "color": "green", "qty": [8, 12, 15] }
+
+For additional information on the operator and method mentioned in this
+section, see the following resources: 
+
+- :manual:`Positional $ Operator </reference/operator/update/positional>` Server Manual Entry
+- :java-core-api:`inc() <com/mongodb/client/model/Updates.html#inc(java.lang.String,java.lang.Number)>` API Documentation
+
+All Array Elements
+~~~~~~~~~~~~~~~~~~
+
+To update all elements in an array that match your query filter, use the
```

Comment:
```suggestion
To update all elements in an array, use the
```

Issue: I don't believe the all positional operator is related to the query filter in the find operation (I say query filter in the find operation as the previous example states that the positional $ operator "updates the first element that matches your query filter" and is referring to the query filter in the find operation). 

I believe that in the example below, the query is "_id : 1", this query does not match an array (the query in the positional $ example must match an array and so queries the quantity field).